firmware/examples/Post/source/main.cpp

```diff
@@ -0,0 +1,287 @@
+  Status status;
+  printf("++++++++++++++++++++++++++++++++++++++\n\n");
+  printf("Beginning I2c scan...\n\n");
+  for (uint8_t address = kFirstI2cAddress; address < kLastI2cAddress; address++)
+  {
+    status = i2c.Write(address, nullptr, 0);
+    if (status == Status::kSuccess)
+    {
+      LOG_INFO("    Found device at address: 0x%02X", address);
+    }
+  }
+  bool gesture_pass       = GestureSensorTest();
+  bool accelerometer_pass = AccelerometerTest();
+  bool temp_pass          = TemperatureSensorTest();
+  bool final_result       = gesture_pass && accelerometer_pass && temp_pass;
+  return final_result;
+}
+
+bool GpioFunctionalityTest()
+{
+  // P0_2 and P0_3 do not need to be tested.
+  // P0_29 and P0_30 do not need to be tested (lack of pullup/pulldown)
+  const bool LQFP80[5][32] = {
+    { 1, 1, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 0, 0, 0, 1,
+      1, 1, 1, 0, 0, 0, 1, 0, 0, 1, 1, 0, 0, 0, 0, 0 },
+    { 1, 1, 0, 0, 1, 0, 0, 0, 1, 1, 1, 0, 0, 0, 1, 1,
+      0, 0, 1, 1, 1, 0, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1 },
+    { 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0,
+      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 },
+    { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 },
+    { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0 }
+  };
+
+  printf("++++++++++++++++++++++++++++++++++++++\n\n");
+  printf("Begin Gpio test...\n\n");
+
+  uint8_t num_ports                   = 5;
+  uint8_t num_pins                    = 32;
+  uint8_t gpio_total                  = 0;
+  uint8_t gpio_passed                 = 0;
+  bool gpio_test[num_ports][num_pins] = { 0 };
+
+  for (uint8_t port = 0; port < num_ports; port++)
+  {
+    for (uint8_t pin = 0; pin < num_pins; pin++)
+    {
+      if (LQFP80[port][pin])
+      {
+        Gpio test_pin(port, pin);
+        test_pin.SetAsOutput();
+        test_pin.SetLow();
+        if (test_pin.Read() == false)
+        {
+          gpio_test[port][pin] = true;
+        }
+        else
+        {
+          LOG_INFO("Pin[%u][%u] Value:%i", port, pin, test_pin.Read());
+          gpio_test[port][pin] = false;
+          if (LQFP80[port][pin] == true && gpio_test[port][pin] == false)
+          {
+            LOG_INFO("Gpio[%u][%u] failed at stage 1.", port, pin);
+          }
+        }
+        test_pin.SetHigh();
+        if (test_pin.Read() && gpio_test[port][pin])
+        {
+          gpio_passed++;
+        }
+        else
+        {
+          LOG_INFO("Pin[%u][%u] Value:%i", port, pin, test_pin.Read());
+          gpio_test[port][pin] = false;
+          if (LQFP80[port][pin] == true && gpio_test[port][pin] == false)
+          {
+            LOG_INFO("Gpio[%u][%u] failed at stage 2.", port, pin);
+          }
+        }
+        gpio_total++;
+      }
+    }
+  }
+  LOG_INFO("Number of Gpio pins that passed: %u/%u", gpio_passed, gpio_total);
+  return gpio_passed / gpio_total;
```

Comment:
```suggestion
  return gpio_passed == gpio_total;
```
Why are you doing it this way? Why not just check that they are equal? Why use divide? This could, potentially result in some problems if there are rounding errors during this calculation.